docker/nginx/conf.d/include/location-skylink

```diff
@@ -0,0 +1,98 @@
+include /etc/nginx/conf.d/include/cors;
+include /etc/nginx/conf.d/include/proxy-buffer;
+include /etc/nginx/conf.d/include/proxy-cache-downloads;
+include /etc/nginx/conf.d/include/track-download;
+
+# redirect purge calls to separate location
+error_page 462 = @purge;
+if ($request_method = PURGE) {
+    return 462;
+}
+
+limit_conn downloads_by_ip 100; # ddos protection: max 100 downloads at a time
+
+# $skylink_v1 and $skylink_v2 variables default to the same value but in case the requested skylink was:
+# a) skylink v1 - it wouldn't matter, no additional logic is executed
+# b) skylink v2 - in a lua block below we will resolve the skylink v2 into skylink v1 and update 
+#      $skylink_v1 variable so then the proxy request to skyd can be cached in nginx (proxy_cache_key 
+#      in proxy-cache-downloads includes $skylink_v1 as a part of the cache key)
+set $skylink_v1 $skylink;
+set $skylink_v2 $skylink;
+
+# variable for Skynet-Proof header that we need to inject 
+# into a response if the request was for skylink v2
+set $skynet_proof '';
+
+# default download rate to unlimited
+set $limit_rate 0;
+
+access_by_lua_block {
+    local httpc = require("resty.http").new()
+
+    -- detect whether requested skylink is v2
+    local isBase32v2 = string.len(ngx.var.skylink) == 55 and string.sub(ngx.var.skylink, 0, 2) == "04"
+    local isBase64v2 = string.len(ngx.var.skylink) == 46 and string.sub(ngx.var.skylink, 0, 2) == "AQ"
+    
+    if isBase32v2 or isBase64v2 then
+        -- 10.10.10.10 points to sia service (alias not available when using resty-http)
+        local res, err = httpc:request_uri("http://10.10.10.10:9980/skynet/resolve/" .. ngx.var.skylink_v2, {
+            headers = { ["User-Agent"] = "Sia-Agent" }
+        })
+
+        -- print error and exit with 500 or exit with response if status is not 200
+        if err or res and res.status ~= ngx.HTTP_OK then
+            ngx.status = err and ngx.HTTP_INTERNAL_SERVER_ERROR or res.status
+            ngx.header["content-type"] = "text/plain"
+            ngx.say(err or res.body)
+            return ngx.exit(err and ngx.HTTP_INTERNAL_SERVER_ERROR or res.status)
```

Comment:
not sure why we don't return res.status here if it's set before defaulting to 500 if an err was defined?

Reply:
yeah that totally makes sense